modules/contracts/apps/challenge/src/handlers/execute.rs

```diff
@@ -0,0 +1,475 @@
+                    next_check_in_by: env.block.height + 777,
+                    metadata,
+                };
+                DAILY_CHECK_INS.save(deps.storage, challenge_id, &check_in)?;
+                return Ok(Response::new()
+                    .add_attribute("action", "check_in")
+                    .add_attribute(
+                        "message",
+                        "You missed the deadline for checking in. You have been given a strike.",
+                    ));
+            }
+            // The admin is checking in before the next check in time, so we can proceeed.
+            block if block < check_in.next_check_in_by => {
+                // this could be configurable, for now
+                // we set the next check in to be 777 blocks from now
+                let next_check_in_block = env.block.height + 777;
+                let check_in = CheckIn {
+                    last_checked_in: env.block.height,
+                    next_check_in_by: next_check_in_block,
+                    metadata,
+                };
+
+                DAILY_CHECK_INS.save(deps.storage, challenge_id, &check_in)?;
+                return Ok(Response::new().add_attribute("action", "check_in"));
+            }
+            _ => {
+                return Err(AppError::Std(StdError::generic_err(
+                    "Something went wrong with the check in.",
+                )));
+            }
+        }
+    } else {
+        return Err(AppError::Std(StdError::generic_err(
+            "Something went wrong with the check in.",
+        )));
+    }
+}
+
+fn cast_vote(
+    deps: DepsMut,
+    env: Env,
+    _info: MessageInfo,
+    _app: &ChallengeApp,
+    vote: Vote<String>,
+    challenge_id: u64,
+) -> AppResult {
+    let vote = vote.check(deps.as_ref())?;
+    let now = env.block.time.seconds();
+    let challenge = CHALLENGE_LIST.load(deps.storage, challenge_id.clone())?;
+    match challenge.status {
+        ChallengeStatus::Active => {}
+        ChallengeStatus::OverAndPending => {}
+        _ => {}
+    }
+
+    let vote = vote.optimisitc();
+    // check that the vote.voter has note has not voted
+    if VOTES
+        .may_load(
+            deps.storage,
+            (challenge_id.to_owned(), vote.voter.to_owned()),
+        )
+        .map_or(false, |votes| votes.iter().any(|v| v.voter == vote.voter))
+    {
+        return Err(AppError::AlreadyVoted {});
+    } else {
+        VOTES.save(
+            deps.storage,
+            (challenge_id.to_owned(), vote.voter.to_owned()),
+            &vote,
+        )?;
+    }
+    Ok(Response::new().add_attribute("action", "cast_vote"))
+}
+
+fn count_votes(
+    deps: DepsMut,
+    info: MessageInfo,
+    _env: Env,
+    app: &ChallengeApp,
+    challenge_id: u64,
+) -> AppResult {
+    let votes = VOTES.may_load(
+        deps.storage,
+        (challenge_id.to_owned(), info.sender.to_owned()),
+    )?;
+
+    let any_false_vote = votes.iter().any(|v| v.approval == Some(false));
```

Comment:
@Buckram123 By the way the tally happens here, we simply check if there is a false vote present. If there is then it's a failed challenge. This was the simplest possible strategy. 

Other tally strategies could be chosen and set in the `ChallengeEntry`, then the associated tally function called.

Reply:
Yup! For that strategy the tally function could be called directly after, saves a tx.